imap/conversations.c

```diff
@@ -1799,8 +1807,12 @@ static int _thread_datesort(const void **a, const void **b)
 {
     const conv_thread_t *ta = (const conv_thread_t *)*a;
     const conv_thread_t *tb = (const conv_thread_t *)*b;
+    struct timespec ta_internaldate, tb_internaldate;
+
+    TIMESPEC_FROM_NANOSEC(&ta_internaldate, ta->internaldate);
+    TIMESPEC_FROM_NANOSEC(&tb_internaldate, tb->internaldate);
 
-    int r = (ta->internaldate - tb->internaldate);
+    int r = (ta_internaldate.tv_sec - tb_internaldate.tv_sec);
```

Comment:
Do we want to sub-sort by nanoseconds as well if r == 0?  It's still going to be stable and better than GUID comparison

Reply:
Yeah, just comparing the nanosec since epoch is stable and deterministic.  Fixed